test/line_number_cache_test.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library dartdoc.cache_test;
+
+import 'dart:io';
+
+import 'package:dartdoc/src/line_number_cache.dart';
+import 'package:path/path.dart' as pathLib;
+import 'package:test/test.dart';
+import 'package:dartdoc/src/tuple.dart';
+
+void main() {
+  group('Verify basic line cache behavior', () {
+    Directory _tempDir;
+
+    setUp(() {
+      _tempDir = Directory.systemTemp.createTempSync('line_number_cache');
+    });
+
+    tearDown(() {
+      _tempDir.deleteSync(recursive: true);
+    });
+
+    test('validate empty file behavior', () {
+      File emptyFile = File(pathLib.join(_tempDir.path, 'empty_file'))
+        ..writeAsStringSync('');
+      LineNumberCache cache = LineNumberCache();
+      expect(cache.lineAndColumn(emptyFile.path, 0), equals(Tuple2(1, 0)));
+    });
+
+    test('single line without newline', () {
+      File singleLineWithoutNewline =
+          File(pathLib.join(_tempDir.path, 'single_line'))
```

Comment:
It hardly matters but I'm a little surprised to see `new`s elided here (and not elsewhere).

Reply:
`new` is now consistently absent.